### PR TITLE
feat(Pydantic): added parameters in pydantic plugin to support strict validation and all the `model_dump` args

### LINKED ISSUE #3572: Enhancement: Support configuring Pydantic's `.model_validate(..., strict=True`)
### Description

Any usage of Strict types has no effect on model validation in Litestar. This is due to the explicit strict=Fasle usage in the pydantic v2's validation method.

### URL to code causing the issue

https://github.com/litestar-org/litestar/blob/65f01ea8ed1162c564b552d637c47dc4220d1cb9/litestar/contrib/pydantic/pydantic_init_plugin.py#L51

### MCVE

```python
from litestar import Litestar, get
from pydantic import BaseModel, StrictBool


class TestModel(BaseModel):
  test_bool: StrictBool


@post("/")
async def test_strict_model(data: TestModel) -> str:
 return "Hello, world!"


app = Litestar([hello_world])
```

### Steps to reproduce

```bash
1. Run the attached MCVE
2. Send a POST req to the server with the following data

{
  "test_bool" : "YES"
}

3. You will receive a success response instead of an error.
```

### Screenshots

```bash
"![SCREENSHOT_DESCRIPTION](SCREENSHOT_LINK.png)"
```

### Logs

_No response_

### Litestar Version

2.9.0
But this is a day 0 bug with Pydantic v2 support afiak.

### Platform

- [ ] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3572">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3572/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by Anu-cool-007**
Combines the following stale PRs:

- #3573
- #3327

Closes #3572 and #3147 

Context: https://github.com/litestar-org/litestar/pull/3327#issuecomment-2171148195

### CODE REVIEW

File: litestar/contrib/pydantic/__init__.py [hunk 1]
```diff
@@ -43,15 +44,45 @@ def _model_dump_json(model: BaseModel | BaseModelV1, by_alias: bool = False) ->
...
 
         Args:
+            exclude: ``type_encoders`` will exclude specified fields
+            exclude_defaults: ``type_encoders`` will exclude default fields
+            exclude_none: ``type_encoders`` will exclude ``None`` fields
+            exclude_unset: ``type_encoders`` will exclude not set fields
+            include: ``type_encoders`` will include only specified fields
+            include: ``type_encoders`` will include only specified fields
             prefer_alias: OpenAPI and ``type_encoders`` will export by alias
+            validate_strict: Pydantic 2.x models will use ``strict=True`` when calling ``.model_validate``
```
Comment (provinzkraut) [inline litestar/contrib/pydantic/__init__.py:77]: These should be the same docstring as in `PydanticInitPlugin`

**Review by provinzkraut — CHANGES_REQUESTED** [review]
Thanks @Anu-cool-007!

**Comment by provinzkraut** [comment]
@kedod can you review this one as well?

**Comment by Anu-cool-007** [comment]
@provinzkraut, is there a specific format for docstrings that I should use?

File: litestar/contrib/pydantic/__init__.py [hunk 2]
```diff
@@ -43,15 +44,43 @@ def _model_dump_json(model: BaseModel | BaseModelV1, by_alias: bool = False) ->
...
+    ) -> None:
+        """Pydantic Plugin to support serialization / validation of Pydantic types / models
+
+        :param exclude: specified fields to exclude when serializing models
+        :param exclude_defaults: Whether to exclude fields which are equal to their default values when serializing models
+        :param exclude_none: Whether to exclude fields with ``value=None`` when serializing models
+        :param exclude_unset: Whether to fields which are not explicitly set when serializing models
+        :param include: specified fields to include when serializing models
+        :param prefer_alias: Whether to use the ``by_alias=True`` flag when serializing models
+        :param validate_strict: Whether to use ``strict=True`` when calling ``.model_validate`` on Pydantic 2.x models
```
Comment (provinzkraut) [inline litestar/contrib/pydantic/__init__.py:75]:
```suggestion
        :param exclude: Fields to exclude during serialization
        :param exclude_defaults: Fields to exclude during serialization when they are set to their default value
        :param exclude_none: Fields to exclude during serialization when they are set to ``None``
        :param exclude_unset: Fields to exclude during serialization when they arenot set
        :param include: Fields to exclude during serialization
        :param prefer_alias: Use the ``by_alias=True`` flag when dumping models
        :param validate_strict: Use ``strict=True`` when calling ``.model_validate`` on Pydantic 2.x models
```

File: litestar/contrib/pydantic/pydantic_init_plugin.py [hunk 3]
```diff
@@ -123,36 +125,116 @@ def extract(annotation: Any, default: Any) -> Any:
...
+    ) -> None:
+        """Pydantic Plugin to support serialization / validation of Pydantic types / models
+
+        :param exclude: specified fields to exclude when serializing models
+        :param exclude_defaults: Whether to exclude fields which are equal to their default values when serializing models
+        :param exclude_none: Whether to exclude fields with ``value=None`` when serializing models
+        :param exclude_unset: Whether to fields which are not explicitly set when serializing models
+        :param include: specified fields to include when serializing models
+        :param prefer_alias: Whether to use the ``by_alias=True`` flag when serializing models
+        :param validate_strict: Whether to use ``strict=True`` when calling ``.model_validate`` on Pydantic 2.x models
```
Comment (provinzkraut) [inline litestar/contrib/pydantic/pydantic_init_plugin.py:156]:
```suggestion
        :param exclude_defaults: Fields to exclude during serialization when they are set to their default value
        :param exclude_none: Fields to exclude during serialization when they are set to ``None``
        :param exclude_unset: Fields to exclude during serialization when they arenot set
        :param include: Fields to exclude during serialization
        :param prefer_alias: Use the ``by_alias=True`` flag when dumping models
        :param validate_strict: Use ``strict=True`` when calling ``.model_validate`` on Pydantic 2.x models
```

**Review by provinzkraut — CHANGES_REQUESTED** [review]
@Anu-cool-007 I went over the docstrings and made some suggestions. Aside from that this is good to go

**Review by provinzkraut — APPROVED** [review]

**Comment by Anu-cool-007** [comment]
@all-contributors add @Anu-cool-007 for code

**Pull request merged**